src/modal/modal-ref.ts

```diff
@@ -34,7 +34,7 @@ export class NgbActiveModal {
 /**
  * A reference to the newly opened modal returned by the `NgbModal.open()` method.
  */
-export class NgbModalRef {
+export class NgbModalRef<T = any> {
```

Comment:
@divdavem : it won't change the behavior or impact how people are using the feature, hence that I used `T = any`
The reason behind `T` is to allow people to do this:
```
currentOpenedDialog: NgbModalRef<ComponentClass>;

// then later:
this.currentOpenedDialog.componentInstance.<auto-complete-here> // if ComponentClass wasn't provided above then it will remain any as it always been and no auto-complete nor type-check will run here.
// this is to reduce common mistakes because of none-typed variable
```

I can remove it but it's overhead efforts, please let me know if it really need another PR just for this 2 lines modification? This one and the one referenced below.

Reply:
So yes, it is intended to be defaulted to `any` so that it wouldn't be required (for those who already developed the feature and already have a working product wouldn't need to go back and provide a `type`.

Reply:
Please let me know if you still need this change on a separate PR, but please be mindful when you decide this. Also please note that it's already mentioned in the PR title and commit message

Reply:
This is part of the general rules for contributions:
https://github.com/ng-bootstrap/ng-bootstrap/wiki/Contributions%3A-How-to-get-a-PR-merged-fast%3F#general-rules

> Please try to keep your PR as small as possible (like really small). You can do the following to avoid big diffs that makes PRs hard to review and integrate:
>  - send separate PRs for separate changes (example: don't mix bug fixes with refactoring or typo fixes; have just one bug fix / feature in a PR);

Reply:
Done